src/sagemaker/remote_function/core/pipeline_variables.py

```diff
@@ -325,27 +333,12 @@ def convert_pipeline_variables_to_pickleable(func_args: Tuple, func_kwargs: Dict
 
     from sagemaker.workflow.entities import PipelineVariable
 
-    from sagemaker.workflow.execution_variables import ExecutionVariables
-
     from sagemaker.workflow.function_step import DelayedReturn
 
-    # Notes:
-    # 1. The s3_base_uri = s3_root_uri + pipeline_name, but the two may be unknown
-    # when defining function steps. After step-level arg serialization,
-    # it's hard to update the s3_base_uri in pipeline compile time.
-    # Thus set a placeholder: _S3BaseUriIdentifier, and let the runtime job to resolve it.
-    # 2. For saying s3_root_uri is unknown, it's because when defining function steps,
-    # the pipeline's sagemaker_session is not passed in, but the default s3_root_uri
-    # should be retrieved from the pipeline's sagemaker_session.
     def convert(arg):
         if isinstance(arg, DelayedReturn):
             return _DelayedReturn(
-                uri=[
-                    _S3BaseUriIdentifier(),
-                    ExecutionVariables.PIPELINE_EXECUTION_ID._pickleable,
-                    arg._step.name,
-                    "results",
-                ],
+                uri=get_step(arg)._properties.OutputDataConfig.S3OutputPath._pickleable,
```

Comment:
The current inconsistent we’re trying to address is that SDK on local side is newer than job container. What about the opposite? What if the SDK on local is older? This is also quite plausible and this change seems breaking

Reply:
I kept the old deserialize logics to resolve such issue.
Such unit tests are kept to verify this case: `What if the SDK on local is older? `
https://github.com/aws/sagemaker-python-sdk/pull/4372/files#diff-84f43d650bd19ebe5e74211c19e9d1a347eea14adf3ae76489ba192f27eebc09R178